### PR TITLE
ios websocketServer support

### PR DESCRIPTION
**Pull request by novain1102**
Re: #https://github.com/cocos/cocos-engine/issues/12222

rely on: https://github.com/cocos/cocos-engine-external/pull/281
### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/bindings/manual/jsb_websocket_server.cpp [hunk 1]
```diff
@@ -34,36 +34,33 @@
...
 
-static bool WebSocketServer_constructor(se::State &s) {
+static bool WebSocketServer_constructor(se::State &s) { // NOLINT(readability-identifier-naming)
     const auto &args = s.args();
-    int argc = (int)args.size();
+    int argc = static_cast<int>(args.size());
     if (argc == 0) {
         se::Object *obj = s.thisObject();
-        WSSPTR cobj = ccnew std::shared_ptr<WebSocketServer>(ccnew WebSocketServer());
+        auto *cobj = ccnew std::shared_ptr<cc::network::WebSocketServer>(ccnew cc::network::WebSocketServer());
```
Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_websocket_server.cpp]: Private data is nolonger needed to be wrapped with `shared_ptr` manually. PrivateObject will do this automatically.
```c++
auto *cobj = ccnew cc::network::WebSocketServer();
obj->setPrivateData(cobj);
```

Accessing with 
```c++
    auto *cobj = static_cast<WebSocket*>(s.nativeThisObject());
```

File: native/cocos/bindings/manual/jsb_websocket_server.cpp [hunk 2]
```diff
@@ -182,17 +179,17 @@ static bool WebSocketServer_onconnection(se::State &s) {
         }
 
         se::Object *obj = se::Object::createObjectWithClass(__jsb_WebSocketServer_Connection_class);
-        WSCONNPTR prv = ccnew std::shared_ptr<WebSocketServerConnection>(conn);
+        auto *prv = ccnew std::shared_ptr<cc::network::WebSocketServerConnection>(conn);
```
Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_websocket_server.cpp]: Use `c::network::WebSocketServerConnection*` as private data directly

File: native/cocos/bindings/manual/jsb_websocket_server.cpp (hunk 1 above)
Comment (novain1102) [reply native/cocos/bindings/manual/jsb_websocket_server.cpp]: done. please have a check.

File: native/cocos/bindings/manual/jsb_websocket_server.cpp (hunk 2 above)
Comment (novain1102) [reply native/cocos/bindings/manual/jsb_websocket_server.cpp]: done. please have a check.

File: native/cocos/bindings/manual/jsb_websocket_server.cpp [hunk 3]
```diff
@@ -34,44 +34,41 @@
...
 
-using namespace cc;
-using namespace cc::network;
+se::Class *__jsb_WebSocketServer_class = nullptr; // NOLINT
+se::Class *__jsb_WebSocketServer_Connection_class = nullptr; // NOLINT
 
-se::Class *__jsb_WebSocketServer_class = nullptr;
-se::Class *__jsb_WebSocketServer_Connection_class = nullptr;
+using WSSPTR = std::shared_ptr<cc::network::WebSocketServer> *;
+using WSCONNPTR = std::shared_ptr<cc::network::WebSocketServerConnection> *;
```
Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_websocket_server.cpp]:
```c++
using WSSPTR = cc::network::WebSocketServer*;
using WSCONNPTR = cc::network::WebSocketServerConnection*;
```

File: native/cocos/bindings/manual/jsb_websocket_server.cpp [hunk 4]
```diff
@@ -117,15 +114,15 @@ static bool WebSocketServer_listen(se::State &s) {
         }
         if (funObj) {
             s.thisObject()->setProperty("__onlisten", se::Value(funObj));
-            std::weak_ptr<WebSocketServer> serverWeak = *cobj;
-            arg_callback = [serverWeak](const ccstd::string &err) {
+            std::weak_ptr<cc::network::WebSocketServer> serverWeak = *cobj;
```
Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_websocket_server.cpp]: Get underlying shared_ptr
```c++

auto *privateObj = s.thisObject()->getPrivateObject();
assert(privateobj->isSharedPtr());
std::weak_ptr<cc::network::WebSocketServer> serverWeak = static_cast<se::SharedPrivateObject*>(privateObj)->getData();

```

File: native/cocos/bindings/manual/jsb_websocket_server.cpp [hunk 5]
```diff
@@ -147,31 +144,31 @@ static bool WebSocketServer_listen(se::State &s) {
         }
     }
 
-    WebSocketServer::listenAsync(*cobj, arg_port, arg_host, arg_callback);
+    cc::network::WebSocketServer::listenAsync(*cobj, argPort, argHost, argCallback);
```
Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_websocket_server.cpp]: Need to root the server js object when successfully listen a port and unroot on close

**Review by minggo — APPROVED** [review]

File: native/cocos/bindings/manual/jsb_websocket_server.cpp (hunk 3 above)
Comment (novain1102) [reply native/cocos/bindings/manual/jsb_websocket_server.cpp]: already removed.

File: native/cocos/bindings/manual/jsb_websocket_server.cpp (hunk 4 above)
Comment (novain1102) [reply native/cocos/bindings/manual/jsb_websocket_server.cpp]: done.

File: native/cocos/bindings/manual/jsb_websocket_server.cpp (hunk 5 above)
Comment (novain1102) [reply native/cocos/bindings/manual/jsb_websocket_server.cpp]: already root/unroot in constructor and onconnection

**Review by PatriceJiang — APPROVED** [review]

**Pull request merged**